devito/builtins.py

```diff
@@ -53,6 +54,135 @@ def smooth(f, g, axis=None):
         dv.Operator(dv.Eq(f, g.avg(dims=axis)), name='smoother')()
 
 
+def gaussian_smooth(f, sigma=1, _order=4, mode='reflect'):
+    """
+    Gaussian smooth function.
+    """
+    class DataDomain(dv.SubDomain):
+
+        name = 'datadomain'
+
+        def __init__(self, lw):
+            super(DataDomain, self).__init__()
+            self.lw = lw
+
+        def define(self, dimensions):
+            return {d: ('middle', self.lw, self.lw) for d in dimensions}
+
+    def fill_data(g, f, dim, lw, mode):
+
+        sl = slice(lw, -lw, 1)
+        indices = ()
```

Comment:
any time you find yourself adding items to a collections, use list/set (these are mutable) and the relevant methods append/add

tuples are immutable and with `+= ` you're merely creating a new object

the effect is the same, but the latter is conceptually wrong

what you could do, once you're done populating your collection, you could cast it to tuple (or frozenset) before returning to the caller

This applies here and pretty much everywhere below